File: Low Level Design (LLD)/DesignPatterns/3. Structural/Chain of Responsibility/Java/ManagerSupportHandler.java
public class ManagerSupportHandler extends SupportHandler {

    @Override
    public void handleRequest(SupportRequest request) {
        if (request.getType().equalsIgnoreCase("advanced")) {
            System.out.println("ManagerSupportHandler: Handling advanced request.");
        } else {
            System.out.println("ManagerSupportHandler: Cannot handle request of type " + request.getType());
        }
    }
}

File: Low Level Design (LLD)/DesignPatterns/3. Structural/Chain of Responsibility/Cpp/src/ManagerSupportHandler.cpp
#include "../include/ManagerSupportHandler.h"
#include <iostream>

void ManagerSupportHandler::handleRequest(const SupportRequest& request) {
    if (request.getType() == "advanced") {
        std::cout << "ManagerSupportHandler: Handling advanced request.\n";
    } else {
        std::cout << "ManagerSupportHandler: Cannot handle request of type " << request.getType() << "\n";
    }
}
